lib/src/init.cpp

```diff
@@ -209,6 +179,45 @@ namespace PIC_PLUS_PLUS {
 		return JSON;
 	}
 
+	void Init::initializeLinearPositions(const int numParticles, std::vector<double>& particlePositions, const double chargeCloudWidth) {
```

Comment:
for all of these helper functions, you should move the parameter that is getting updated all the way to the left and prefix them with `inOut` so that it is easier to see that they are inplace updates